### PR TITLE
Feature/lazy imports

### PR DESCRIPTION
**Pull request by samwelkanda**
Adds a utility method `lazy_import` that will perform lazy loading of module imports.

### CODE REVIEW

File: kiota_abstractions/utils.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def lazy_import(name):
+    """Lazily imports a python module given its absolute path
+
+    Args:
+        name (str): Absolute path to the module
+
+    Returns:
+        module: The module to be imported
+    """
+    spec = importlib.util.find_spec(name)
```
Comment (baywet) [inline kiota_abstractions/utils.py:14]: there should probably be some defensive programing in place

File: kiota_abstractions/utils.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import importlib.util
+import sys
+
+
+def lazy_import(name):
+    """Lazily imports a python module given its absolute path
```
Comment (baywet) [inline kiota_abstractions/utils.py:6]: We should probably indicate in the doc comment that this method is only meant for the generated code and that people should not take a dependency on it outside of that scenario.

File: kiota_abstractions/utils.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    Args:
+        name (str): Absolute path to the module
+
+    Returns:
+        module: The module to be imported
+    """
+    spec = importlib.util.find_spec(name)
+
+    loader = importlib.util.LazyLoader(spec.loader)
```
Comment (baywet) [inline kiota_abstractions/utils.py:16]: the case where the loader or module cannot be found should probably be handled as well.

**Review by baywet — CHANGES_REQUESTED** [review]

File: kiota_abstractions/utils.py (hunk 1 above)
Comment (samwelkanda) [reply kiota_abstractions/utils.py:20]: Thanks. Done in https://github.com/microsoft/kiota-abstractions-python/pull/19/commits/aa4c29726a09d763c66181f694dd13c3bb7be701

File: kiota_abstractions/utils.py (hunk 2 above)
Comment (samwelkanda) [reply kiota_abstractions/utils.py]: Added in https://github.com/microsoft/kiota-abstractions-python/pull/19/commits/aa4c29726a09d763c66181f694dd13c3bb7be701

File: kiota_abstractions/utils.py (hunk 3 above)
Comment (samwelkanda) [reply kiota_abstractions/utils.py:25]: As long as the spec is found, any further errors with loader and module will be gracefully handled by `importlib`.

**Review by baywet — APPROVED** [review]

**Pull request merged**